EventListener/Login.php

```diff
@@ -63,6 +67,28 @@ public function setCustomerId($value)
         $this->options['customerId'] = $value;
     }
 
+    /**
+     * Return an ID or the Username if no ID available.
+     *
+     * @param TokenInterface $token
+     *
+     * @return mixed
+     */
+    protected function getUserIdentifier(TokenInterface $token)
+    {
+        $tokenUser = $token->getUser();
+
+        if ($tokenUser instanceof eZUser || !$tokenUser instanceof UserInterface) {
+            return $tokenUser->getAPIUser()->id;
+        }
+
+        if (method_exists($tokenUser, 'getId')) {
+            return $tokenUser->getId();
+        }
+
+        return $tokenUser->getUsername();
```

Comment:
We have more to do here, the repository Id is clearly not the UserId in case of External provider
That is just the API User Id.

We have the session inject then I think it should be ok.

BUT I'm wondering if we should keep this ID management, I mean if Yoochoose is fine with a Username why not just put the Username all the time ?

In a certain way, that was the @lolautruche question.

Is it better to have an ID or could we use the Username all the time ?